src/lib/bcmp/dfu/bm_dfu_core.cpp

```diff
@@ -7,28 +7,28 @@
 #include "bm_dfu.h"
 #include "bm_dfu_client.h"
 #include "bm_dfu_host.h"
+#include "bm_dfu_generic.h"
 #include "task_priorities.h"
 #include "device_info.h"
 #include "lpm.h"
 
-using namespace cfg;
+extern "C" {
+#include "messages.h"
+#include "packet.h"
+}
 
 typedef struct dfu_core_ctx_t {
-    libSmContext_t sm_ctx;
+    LibSmContext sm_ctx;
     bm_dfu_event_t current_event;
     bool pending_state_change;
     uint8_t new_state;
     bm_dfu_err_t error;
     uint64_t self_node_id;
     uint64_t client_node_id;
     bcmp_dfu_tx_func_t bcmp_dfu_tx;
-    NvmPartition *dfu_partition;
     update_finish_cb_t update_finish_callback;
-    Configuration* sys_cfg;
```

Comment:
these are no longer needed since the bm_dfu_wrapper file should be the only dfu file accessing the configs/nvm partition